test/gtest/uct/test_md.cc

```diff
@@ -299,11 +301,26 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
 
         /* test mem_query API */
         uct_md_mem_attr_t mem_attr;
-        mem_attr.field_mask = UCT_MD_MEM_ATTR_FIELD_MEM_TYPE |
-                              UCT_MD_MEM_ATTR_FIELD_SYS_DEV;
-        status = uct_md_mem_query(md(), address, buffer_size, &mem_attr);
-        ASSERT_UCS_OK(status);
-        EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
+        mem_attr.field_mask = UCT_MD_MEM_ATTR_FIELD_MEM_TYPE     |
+                              UCT_MD_MEM_ATTR_FIELD_SYS_DEV      |
+                              UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS |
+                              UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH;
+
+        for (unsigned i = 0; i < 300; i++) {
+            offset        = ucs::rand() % buffer_size;
+            length_offset = ucs::rand() % buffer_size;
```

Comment:
will it work if length_offset==0?
also rename
offset -> slice_offset
length -> slice_length

Reply:
I see that we require uct_md_mem_query to be provided with non-zero lengths as per UCT API documentation. Shall I skip if length_offset==0?

Reply:
yes, or add 1 to rand result